dask/dataframe/categorical.py

```diff
@@ -115,7 +115,6 @@ def categorize(df, columns=None, index=None, split_every=None, **kwargs):
 
     prefix = 'get-categories-agg-' + token
     k = df.npartitions
-    b = a
```

Comment:
`b` either gets redefined a few lines below in the `while` loop, or is never used.

Reply:
Is it ever the case that the while loop doesn't trigger?

Reply:
`b` is never used if the while loop doesn't trigger; there are no references to `b` outside the while loop (aside from the deleted `b = a` line).

Breaking down the two scenarios:  If the while loop triggers, `b` immediately gets redefined on the first line within the while loop, so the `b = a` line is redundant and can be deleted.  If the while loop doesn't trigger, `b` is never used, so the `b = a` line is redundant and can be deleted.